database/engine/journalfile.c

```diff
@@ -162,19 +189,21 @@ int create_journal_file(struct rrdengine_journalfile *journalfile, struct rrdeng
 
     ret = uv_fs_write(NULL, &req, file, &iov, 1, 0, NULL);
     if (ret < 0) {
-        fatal("uv_fs_write: %s", uv_strerror(ret));
-    }
-    if (req.result < 0) {
-        fatal("uv_fs_write: %s", uv_strerror((int)req.result));
+        assert(req.result < 0);
+        error("uv_fs_write: %s", uv_strerror(ret));
+        ++ctx->stats.io_errors;
+        rrd_stat_atomic_add(&global_io_errors, 1);
     }
     uv_fs_req_cleanup(&req);
     free(superblock);
```

Comment:
freez/mallocz?

Reply:
mallocz not used on purpose because ?

Reply:
did you look at the code?

Reply:
My question is:
Why use malloc/free in the first place and not use "mallocz/freez" everywhere consistently?

Reply:
I don't use malloc, please look at the code before doing the code review.

Reply:
Oh i see, so we use `posix_memalign()` for performance, got it